src/main/scala/firrtl/passes/CommonSubexpressionElimination.scala

```diff
@@ -2,19 +2,31 @@
 
 package firrtl.passes
 
+import firrtl.Mappers._
 import firrtl._
+import firrtl.annotations.NoTargetAnnotation
 import firrtl.ir._
-import firrtl.Mappers._
-import firrtl.options.Dependency
+import firrtl.options.{Dependency, HasShellOptions, RegisteredTransform, ShellOption}
+
+/** Indicate that CommonSubexpressionElimination should not be run */
+case object NoCommonSubexpressionElimination extends NoTargetAnnotation
 
-object CommonSubexpressionElimination extends Pass {
+object CommonSubexpressionElimination extends Transform with HasShellOptions with DependencyAPIMigration {
 
   override def prerequisites = firrtl.stage.Forms.LowForm
   override def optionalPrerequisiteOf =
     Seq(Dependency[SystemVerilogEmitter], Dependency[VerilogEmitter])
 
   override def invalidates(a: Transform) = false
 
+  val options = Seq(
+    new ShellOption[Unit](
+      longOption = "no-common-subexpression-elimination",
```

Comment:
It's probably fine to keep the option shorter:

```suggestion
      longOption = "no-cse",
```